### ISSUE TITLE
_log_softmax is decomposed as log(softmax(x)), which returns -inf for logits far below the row max where eager returns a finite value

### ISSUE THREAD

**Issue by botify-service** [tenstorrent/pytorch2.0_ttnn#1341]
### Summary

`aten._log_softmax.default` is decomposed into `log(softmax(x))`. That is the composition
`log_softmax` exists to avoid: `softmax` underflows to exactly `0` for any element far enough
below the row max, and `log(0)` is `-inf`, where the true value is an ordinary finite number.
The result is `-inf` in the compiled graph where eager PyTorch returns a normal negative float.

Separately and in the same area, the `_softmax` lowering drops the `half_to_float` argument that
this decomposition takes care to forward.

### 1. The unstable decomposition

`torch_ttnn/passes/lowering/to_tt_pass.py`:

```python
if node.target == torch.ops.aten._log_softmax.default:
    dim, half_to_float = get_arg(node, 1, "dim"), get_arg(node, 2, "half_to_float")
    softmax = g.call_function(torch.ops.aten._softmax.default, args=(args[0], dim, half_to_float))
    log = g.call_function(torch.ops.aten.log.default, args=(softmax,))
    return log
```

PyTorch implements `log_softmax` as `x - logsumexp(x)`, which never forms the probability. This
rewrite forms it, and the probability is the value that leaves representable range.

Evaluated in `float32` with each intermediate rounded — `log(softmax(x))` against
`x - logsumexp(x)`:

| logits | `log(softmax)` | true `log_softmax` |
|---|---|---|
| `[10, 0, -100]` | `-4.54e-05`, `-10`, **`-inf`** | `-0.000045`, `-10.000045`, `-110.000045` |
| `[50, 0, -120]` | `0`, `-50`, **`-inf`** | `0.000000`, `-50.000000`, `-170.000000` |
| `[20, -80, -200]` | `0`, `-99.9831`, **`-inf`** | `0.000000`, `-100.000000`, `-220.000000` |
| `[2, 1, 0.5]` | `-0.464369`, `-1.46437`, `-1.96437` | matches — control row |

`-110`, `-170` and `-220` are ordinary `float32` values. Nothing about the answer is near a
representable limit; only the intermediate probability is. The control row shares the mechanism
and passes, so this is not general breakage of the op.

The threshold is the `exp` underflow point: once `x_i - max` falls below roughly `-88` in
`float32` the probability flushes to zero, and in `bfloat16` the gap needed is smaller still.
A logit gap of 100+ is unremarkable in a trained classifier over a large vocabulary.

### Why this matters more than the usual accuracy report

`log_softmax` is the first half of cross-entropy. `torch.nn.functional.cross_entropy` and
`nll_loss(log_softmax(x), y)` are how essentially every classifier computes its loss. An `-inf`
in the log-probabilities becomes `inf` or `NaN` in the loss and then in every gradient, so the
failure does not stay local — it takes out the training step. And it appears only once the model
becomes confident enough to open a 100-point logit gap, which is to say partway through training
rather than at the start.

### 2. `half_to_float` is forwarded here and dropped downstream

The decomposition above reads `half_to_float` and passes it into `_softmax`. The `_softmax`
lowering then discards it:

```python
if target == torch.ops.aten._softmax.default:
    kwargs = {"numeric_stable": True, **kwargs}
    return self.call_function_prop_meta(ttnn.softmax, args[:2], kwargs)   # args[2] dropped
```

`aten._softmax(self, dim, half_to_float)` takes three positional arguments; `args[:2]` keeps the
first two. `half_to_float` is set when `torch.softmax(x, dim, dtype=torch.float32)` is called on a
half-precision input, and it makes the output `float32`. Dropping it yields a half-precision
output where eager returns `float32`.

Worth noting the internal inconsistency: one site takes care to carry the argument through, the
other discards it. Whichever is intended, they disagree.

### Expected

`aten._log_softmax.default` should lower to a stable formulation — `x - logsumexp(x, dim)` — so
that finite inputs give finite outputs. `ttnn.softmax` already accepts `numeric_stable=True`,
which handles the forward direction's max-subtraction; the missing piece is not forming the
probability before taking the log.

If a dedicated stable path is not available, declining to lower `_log_softmax` and leaving the
aten op in place would be preferable to rewriting it into a form that returns `-inf`.

For `half_to_float`: either forward it into the `ttnn.softmax` call and cast the output
accordingly, or guard the non-default case so the node is not silently rewritten.

### Steps to reproduce

```python
import torch, torch_ttnn, ttnn

class M(torch.nn.Module):
    def forward(self, x):
        return torch.nn.functional.log_softmax(x, dim=-1)

device = ttnn.open_device(device_id=0)
m = M()
# a confident classifier row: one logit far below the max
x = torch.tensor([[10.0, 0.0, -100.0]], dtype=torch.bfloat16)

eager = m(x)
compiled = torch.compile(m, backend=torch_ttnn.backend,
                         options=torch_ttnn.TorchTtnnOption(device=device))(x)

print("eager   :", eager)
print("compiled:", compiled)
print("finite everywhere:", torch.isfinite(compiled).all().item())   # expected True

ttnn.close_device(device)
```

A correct lowering prints finite values on both lines. `[[2.0, 1.0, 0.5]]` passes either way,
which isolates the cause to the logit gap rather than to the op.

### Why the current tests do not catch it

The `log_softmax` lowering tests use small, closely-spaced logits — random inputs in a narrow
range — where the probability never underflows and `log(softmax(x))` agrees with the stable form
to within tolerance. Nothing samples a row with a large gap between the max and the minimum
logit, which is the only condition that triggers it.

### Notes on verification

Source analysis of the lowering pass plus exact arithmetic; I have not run this on hardware. The
decomposition and the dropped `args[2]` are readable at the lines quoted. The table was produced
by evaluating both formulations in `float32` with every intermediate rounded, modelling
flush-to-zero on the underflowing `exp` — it reproduces the `-inf` boundary from `float32`
semantics alone, so it should hold on device, and the repro above would confirm it.
